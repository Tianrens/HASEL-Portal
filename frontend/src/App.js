import React, { useContext } from 'react';
import './App.css';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import LandingPage from './views/pages/LandingPage/LandingPage';
import { AppContext } from './AppContextProvider';
import NewRequest from './views/pages/NewRequest/NewRequest';
import PendingApproval from './views/pages/PendingApproval/PendingApproval';
import SignupPage from './views/pages/SignupPage/SignupPage';

function App() {
    const idToken = useContext(AppContext).firebaseUserIdToken;

    function AuthenticatedPaths() {
        return (
            <Switch>
                <Route exact path='/signup' component={SignupPage} />
                <Route path='/request' component={NewRequest} />
                <Route path='/pending' component={PendingApproval} />
                {/* Default path */}
                <Route component={SignupPage} />
            </Switch>
        );
    }

    function UnauthenticatedPaths() {
        return (
            <Switch>
                <Route exact path='/' component={LandingPage} />
                {/* Default path */}
                <Route component={LandingPage} />
            </Switch>
        );
    }

    return <Router>{idToken ? <AuthenticatedPaths /> : <UnauthenticatedPaths />}</Router>;
}

export default App;
