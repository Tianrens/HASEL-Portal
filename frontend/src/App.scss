@import './assets/colours.module';

.MuiTableBody-root > .MuiTableRow-root:hover,
.MuiMenuItem-root:hover {
    background-color: $subHeading;
    transition: all 0.3s;
}

.MuiPaper-root .MuiTableHead-root {
    background-color: $purple700;
}

.MuiMenu-paper {
    max-height: calc(50% - 35px);
}
