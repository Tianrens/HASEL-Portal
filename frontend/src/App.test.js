import React from 'react';
import renderer from 'react-test-renderer';
import StylesProvider from '@mui/styles/StylesProvider';
import { SnackbarProvider } from 'notistack';
import App from './App';

test('App renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <SnackbarProvider>
                <App />
            </SnackbarProvider>
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
