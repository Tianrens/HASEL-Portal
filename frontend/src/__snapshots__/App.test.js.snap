// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div
  className="fullPage"
>
  <div
    className="heroPattern"
  >
    <div
      className="container"
    >
      <div
        className="content"
      >
        <h1
          className="header"
        >
          HASEL Lab
        </h1>
        <div
          className="QnA"
        >
          <h2
            className="subheader"
          >
            What is HASEL?
          </h2>
          <p
            className="text"
          >
            The Human Aspects in Software Engineering Lab (HASEL) researches projects related to improving software practices and processes, with a focus on the human aspects of software engineering. We have several high performance machine learning computers available for use in the lab. They can be accessed remotely through SSH as well.
          </p>
          <h2
            className="subheader"
          >
            How do I gain access?
          </h2>
          <p
            className="text"
          >
            Login with your university connected Google account below. Then enter your details and apply for a workstation. After access is granted you will receive an email with your workstation details. You can then use this site to book the lab computers.
          </p>
        </div>
      </div>
      <div
        className="expandingSpacer"
      />
      <div
        className="loginContainer"
      >
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root button undefined   css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Login
        </button>
      </div>
    </div>
  </div>
  <div
    className="rightSide"
  >
    <div
      className="bgImg"
    />
  </div>
</div>
`;
